lib/widgets/message_list.dart

```diff
@@ -503,10 +513,52 @@ class MarkAsReadWidget extends StatelessWidget {
                   .merge(weightVariableTextStyle(context, wght: 400))),
                 shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(7)),
               ),
-              onPressed: () => _handlePress(context),
+              onPressed: () => loading ? null : _handlePress(context),
```

Comment:
```suggestion
              onPressed: loading ? null : () => _handlePress(context),
```

See dartdoc:

```dart
  /// Called when the button is tapped or otherwise activated.
  ///
  /// If this callback and [onLongPress] are null, then the button will be disabled.
  ///
  /// See also:
  ///
  ///  * [enabled], which is true if the button is enabled.
  final VoidCallback? onPressed;
```

This should let the `FilledButton` signal to screen-reader software that it's disabled, which is helpful for accessibility.

…I notice that this causes `FilledButton` to apply its own chosen "disabled" style to the button, making it gray, which is very different from Vlad's design. We can tune that style, with [`style.backgroundColor`](https://api.flutter.dev/flutter/material/ButtonStyle/backgroundColor.html), but unfortunately that API doesn't seem to offer a way to animate the transition. So I think we just keep the animation you've added (and keep the null `onPressed` for the disabled state) but add some lines to clobber that default gray style:

```dart
              style: FilledButton.styleFrom(
                // …
                // (remove `backgroundColor: _UnreadMarker.color` and its comment from here)
              ).copyWith(
                // clobber `FilledButton`'s default disabled state
                // TODO(#95) need dark-theme colors (foreground and background)
                foregroundColor: WidgetStateColor.resolveWith((_) => Colors.white),
                backgroundColor: WidgetStateColor.resolveWith((_) => _UnreadMarker.color),
              ),
```